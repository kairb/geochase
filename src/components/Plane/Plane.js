import React from 'react'
import plane from '../../assets/Fly.png';
import deadPlane from '../../assets/Dead.png'
import {makeStyles} from "@material-ui/core";

const useStyles = makeStyles({
  plane:({position}) => ({
    position:"absolute",
    width: "10%",
    top: position,
  })
});


const Plane = ({position, gameOver}) => {
  const classes = useStyles({position});
  return(
    <img src={(gameOver)? deadPlane: plane} alt="plane" className={classes.plane}/>
  )
}

export default Plane;