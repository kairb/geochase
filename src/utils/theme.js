import {createMuiTheme} from '@material-ui/core';

export default createMuiTheme(() => ({
  palette:{
    primary:'green',
    secondary:'white',
  }
}))