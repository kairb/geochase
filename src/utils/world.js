import ground1 from '../assets/scenery/square-1.png';
import ground2 from '../assets/scenery/square-2.png';
import ground3 from '../assets/scenery/square-3.png';
import ground4 from '../assets/scenery/square-4.png';
import ground5 from '../assets/scenery/square-dark.png';

const getRandomGround = () => {
  return `ground${parseInt(Math.random() * 4 + 1)}`;
};

export const createWorld = () => {
  // world goes thisa way
  //  |
  //  |
  //  V
  const getInitialRow = () => {
    return [0, 0, 0, 0, 0, 0, 0, 0, 0, 'ground1'];
  };
  return new Array(40).fill(getInitialRow(), 0, 40);
};

export const blockMap = {
  0: {
    // color: 'none',
  },
  1: {
    // color: 'green',
  },
  ground1: { image: ground1 },
  ground2: { image: ground2 },
  ground3: { image: ground3 },
  ground4: { image: ground4 },
  ground5: { image: ground5 },
};

const getRandom = () => {
  return Math.random() > 0.5 ? "ground5" : 0;
};

export const getCollisionHeight = (world) => {
  const height1 = getColumnWorldHeight(world[18]);
  const height2 = getColumnWorldHeight(world[19]);
  const height3 = getColumnWorldHeight(world[20]);
  const height4 = getColumnWorldHeight(world[21]);
  return [height1, height2, height3, height4].sort()[0];
};

const getColumnWorldHeight = (column) => {
  return ((column.join('').indexOf('ground') - 1) / 10) * window.innerHeight;
};

const getNextBlock = (above, below, index) => {
  if (above !== 0) {
    return 'ground5';
  } else if (below === 0) {
    return 0;
  } else {
    return getRandom();
  }
};

const getNextColumn = (column) => {
  return column
    .map((block, index) => {
      if (index < 2) {
        return 0;
      } else if (index === 9) {
        return 'ground5';
      } else {
        return getNextBlock(column[index - 1], column[index + 1], index);
      }
    })
    .map((block, index, newWorld) => {
      console.log(newWorld);
      if (!newWorld[index - 1] && newWorld[index] && !newWorld[index-2]) {
        return getRandomGround();
      }
      if (newWorld[index - 1] && newWorld[index + 1]) {
        return 'ground5';
      }
      return block;
    });
};

export const updateWorld = (world) => {
  return [...world.slice().splice(1, 39), getNextColumn(world[39])];
};
