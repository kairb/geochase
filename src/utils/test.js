import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider} from '@material-ui/core/styles';
import theme from './theme';
export const renderWithThemeWrapper = (component) =>
  render(
    <ThemeProvider theme={theme}>
      {component}
    </ThemeProvider>
  );