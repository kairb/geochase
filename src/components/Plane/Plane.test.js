import React from 'react';
import {renderWithThemeWrapper} from '../../utils/test';
import Plane from './Plane';

describe('Plane component', ()=>{
  it('should render the alive plane when gameOver is false', () =>{
    const {container} = renderWithThemeWrapper(<Plane position={0} gameOver={false}/>);
    expect(container.querySelector('img').src).toContain("Fly.png")
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the dead plane when gameOver is true', () =>{
    const {container} = renderWithThemeWrapper(<Plane position={0} gameOver={true}/>);
    expect(container.querySelector('img').src).toContain("Dead.png")
  })


})