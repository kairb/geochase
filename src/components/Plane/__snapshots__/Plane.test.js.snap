// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plane component should render the alive plane when gameOver is false 1`] = `
<img
  alt="plane"
  class="makeStyles-plane-1 makeStyles-plane-2"
  src="Fly.png"
/>
`;
